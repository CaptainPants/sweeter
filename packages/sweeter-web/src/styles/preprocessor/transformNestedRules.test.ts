import { parse, transformNestedRules } from './index.js';

it('Nested media query implicitly adds parent selector', () => {
    const parsed = parse(`
        .banana {
            @media screen {
                color: red;
            }
        }
    `);

    const transformed = transformNestedRules(parsed);

    expect(transformed).toMatchSnapshot();
});

it('Normal @ rule nesting', () => {
    const content = `
        @media screen {
            @media (max-width: 200px) {
                a {
                    color: green;
                }

                b {
                    color: green;
                }
            }

            div {
                color: rebeccapurple;

                @media (min-width: 200px) {
                    background: grey;
                }
            }
        }
    `;

    const parsed = parse(content);

    const transformed = transformNestedRules(parsed);

    expect(transformed).toMatchSnapshot();
});
