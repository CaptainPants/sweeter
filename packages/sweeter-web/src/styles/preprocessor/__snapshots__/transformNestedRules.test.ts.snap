// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Nested media query implicitly adds parent selector 1`] = `
[
  {
    "$nodeType": "at",
    "nestedRules": [
      {
        "$nodeType": "rule",
        "nestedRules": [],
        "properties": [
          {
            "$nodeType": "property",
            "name": "color",
            "value": "red",
          },
        ],
        "selectors": [
          ".banana",
        ],
      },
    ],
    "parameters": "screen",
    "type": "media",
  },
]
`;

exports[`Normal @ rule nesting 1`] = `
[
  {
    "$nodeType": "at",
    "nestedRules": [
      {
        "$nodeType": "at",
        "nestedRules": [
          {
            "$nodeType": "rule",
            "nestedRules": [],
            "properties": [
              {
                "$nodeType": "property",
                "name": "color",
                "value": "green",
              },
            ],
            "selectors": [
              "a",
            ],
          },
          {
            "$nodeType": "rule",
            "nestedRules": [],
            "properties": [
              {
                "$nodeType": "property",
                "name": "color",
                "value": "green",
              },
            ],
            "selectors": [
              "b",
            ],
          },
        ],
        "parameters": "(max-width: 200px)",
        "type": "media",
      },
      {
        "$nodeType": "rule",
        "nestedRules": [],
        "properties": [
          {
            "$nodeType": "property",
            "name": "color",
            "value": "rebeccapurple",
          },
        ],
        "selectors": [
          "div",
        ],
      },
      {
        "$nodeType": "at",
        "nestedRules": [
          {
            "$nodeType": "rule",
            "nestedRules": [],
            "properties": [
              {
                "$nodeType": "property",
                "name": "background",
                "value": "grey",
              },
            ],
            "selectors": [
              "div",
            ],
          },
        ],
        "parameters": "(min-width: 200px)",
        "type": "media",
      },
    ],
    "parameters": "screen",
    "type": "media",
  },
]
`;
